src/AzureWriter.cs

```diff
@@ -43,12 +44,12 @@ public sealed class AzureWriter : AzurePreWriter
         /// <see cref="StoreWriter.DoOptCommitAsync"/>
         protected override async Task DoOptCommitAsync(Hash hash, Func<Task> optionalWrite, CancellationToken cancel)
         {
-            var finalBlob = _persistent.GetBlockBlobReference(AzureReadOnlyStore.AzureBlobName(_realm, hash));
+            var finalBlob = _persistent.GetBlobClient(AzureReadOnlyStore.AzureBlobName(_realm, hash));
 
             // Final blob already exists (maybe it was uploaded earlier), do nothing.
-            if (await AzureRetry.OrFalse(() => finalBlob.ExistsAsync(null, null, cancel)).ConfigureAwait(false))
+            if (await AzureRetry.OrFalse(async () => await finalBlob.ExistsAsync(cancellationToken: cancel)).ConfigureAwait(false))
             {
-                _onCommit?.Invoke(_stopwatch.Elapsed, _realm, hash, finalBlob.Properties.Length, true);
+                _onCommit?.Invoke(_stopwatch.Elapsed, _realm, hash, finalBlob.GetProperties()?.Value?.ContentLength ?? 0, true);
```

Comment:
If `GetProperties()?.Value?.ContentLength` are null, it is safer to throw an exception than to continue with an incorrect value of 0.